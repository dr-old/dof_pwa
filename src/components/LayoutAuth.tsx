import { Box, Container, styled, useMediaQuery, useTheme } from "@mui/material";
import React, { ReactNode } from "react";
import ActivityIndicator from "./ActivityIndicator";

interface LoginProps {
  children: ReactNode;
  loading?: boolean;
}

const BoxMain = styled(Box)(({ theme }) => ({
  height: "100%",
  borderRadius: theme.shape.borderRadius * 2.5,
  backgroundColor: theme.palette.primary.main,
}));

const LayoutAuth: React.FC<LoginProps> = ({ children, loading }) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));

  return (
    <React.Fragment>
      {loading && <ActivityIndicator fullScreen={true} />}
      <Box
        sx={{
          height: isMobile ? "100%" : "100vh",
          padding: { xs: 1, sm: 2, md: 3, lg: 4 },
        }}>
        <BoxMain
          sx={{
            padding: { xs: 3, sm: 2, md: 3, lg: 4 },
          }}>
          <Container maxWidth="xl">{children}</Container>
        </BoxMain>
      </Box>
    </React.Fragment>
  );
};

export default LayoutAuth;
