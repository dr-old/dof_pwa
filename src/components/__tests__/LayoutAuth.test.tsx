import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import LayoutAuth from "../LayoutAuth";

// Mock the ActivityIndicator component
jest.mock("../ActivityIndicator", () => {
  return function DummyActivityIndicator() {
    return <div data-testid="activity-indicator" />;
  };
});

describe("LayoutAuth Component", () => {
  it("renders children correctly", () => {
    render(
      <LayoutAuth>
        <div data-testid="child-element">Test Child</div>
      </LayoutAuth>
    );
    expect(screen.getByTestId("child-element")).toBeInTheDocument();
  });

  it("renders the ActivityIndicator when loading is true", () => {
    render(
      <LayoutAuth loading={true}>
        <div>Test Child</div>
      </LayoutAuth>
    );
    expect(screen.getByTestId("activity-indicator")).toBeInTheDocument();
  });

  it("does not render the ActivityIndicator when loading is false", () => {
    render(
      <LayoutAuth loading={false}>
        <div>Test Child</div>
      </LayoutAuth>
    );
    expect(screen.queryByTestId("activity-indicator")).not.toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <LayoutAuth loading={false}>
        <div>Test Child</div>
      </LayoutAuth>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
