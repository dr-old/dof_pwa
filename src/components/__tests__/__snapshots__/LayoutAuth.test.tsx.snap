// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutAuth Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-8k0fxg"
  >
    <div
      class="MuiBox-root css-144j783"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div>
          Test Child
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
